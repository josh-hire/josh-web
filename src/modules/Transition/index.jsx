import PropTypes from 'prop-types';
import { motion, AnimatePresence } from 'framer-motion';
import { useRouter } from 'next/router';

const Transition = ({ children }) => {
  const { pathname } = useRouter();
  const variants = {
    out: {
      opacity: 0,
      y: 0,
      transition: {
        duration: 0,
      },
    },
    in: {
      opacity: 1,
      y: 0,
      transition: {
        duration: 0.5,
        delay: 0.15,
      },
    },
  };
  return (
    <div className="effect-1">
      <AnimatePresence exitBeforeEnter={true} initial={true}>
        <motion.div animate="in" exit="out" initial="out" key={pathname} variants={variants}>
          {children}
        </motion.div>
      </AnimatePresence>
    </div>
  );
};

Transition.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Transition;
