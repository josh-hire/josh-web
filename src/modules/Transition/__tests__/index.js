import Transition from '../';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      slug: 'test transition',
    },
    asPath: 'http://test.com',
  }),
}));

describe('src/modules/Transition', () => {
  test('Transition', () => {
    const { asFragment } = render(
      <Transition>
        <div />
      </Transition>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
