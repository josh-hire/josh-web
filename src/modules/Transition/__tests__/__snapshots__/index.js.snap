// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/modules/Transition Transition 1`] = `
<DocumentFragment>
  <div
    class="effect-1"
  >
    <div
      style="opacity: 0; transform: none;"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
